pkg/systemd/services/service.jsx

```diff
@@ -40,147 +41,143 @@
         console.debug.apply(console, arguments); // not-covered: debugging
 }
 
-export class Service extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            unit_id: null,
-            unit_props: null,
-            error: null,
-            reloading: false,
-        };
-
-        this.path = null;
-        this.subscriptions = [];
-    }
-
-    async componentDidMount() {
-        const dbus = this.props.dbusClient;
-        try {
-            const [path] = await dbus.call(s_bus.O_MANAGER, s_bus.I_MANAGER, "LoadUnit", [this.props.unitId]);
-            this.path = path;
-        } catch (ex) {
-            this.setState({ error: ex.toString() });
-            return;
-        }
-
-        this.subscriptions.push(dbus.subscribe(
+export const Service = ({ dbusClient, owner, unitId, unitIsValid, addTimerProperties, pinnedUnits }) => {
+    const _path = useRef(null);
+    const [error, setError] = useState(null);
+    const [reloading, setReloading] = useState(false);
+    const [unitProps, setUnitProps] = useState(null);
+
+    const updateProperties = () => {
+        const path = _path.current?.path;
+        if (!path)
+            console.error("Service detail", unitId, "updateProperties called without path");
+        const promises = [dbusClient.call(path, s_bus.I_PROPS, "GetAll", [s_bus.I_UNIT])];
+
+        if (unitId.endsWith(".timer"))
+            promises.push(dbusClient.call(path, s_bus.I_PROPS, "GetAll", [s_bus.I_TIMER]));
+        else if (unitId.endsWith(".socket"))
+            promises.push(dbusClient.call(path, s_bus.I_PROPS, "GetAll", [s_bus.I_SOCKET]));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18481-20230313-094909-6117a6e5-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)